lib/core/api/api-factory.js

```diff
@@ -181,7 +181,7 @@
       }
 
       if(this.beforeCreate) {
-        this.beforeCreate(this, data);
+        data = this.beforeCreate(data);
```

Comment:
Now beforeCreate behaves like beforeUpdate.  These two functions had very different signatures for no clear reason.